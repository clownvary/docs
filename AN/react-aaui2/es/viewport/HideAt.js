import _extends from 'babel-runtime/helpers/extends';
import React from 'react';

import Viewport from './Viewport';

function HideAt(props) {
  return React.createElement(Viewport, _extends({ not: true }, props));
}

export default HideAt;