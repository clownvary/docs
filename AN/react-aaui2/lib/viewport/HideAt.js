'use strict';

exports.__esModule = true;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Viewport = require('./Viewport');

var _Viewport2 = _interopRequireDefault(_Viewport);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function HideAt(props) {
  return _react2.default.createElement(_Viewport2.default, (0, _extends3.default)({ not: true }, props));
}

exports.default = HideAt;
module.exports = exports['default'];