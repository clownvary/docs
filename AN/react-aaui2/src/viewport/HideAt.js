import React from 'react'

import Viewport from './Viewport'

function HideAt(props) {
  return <Viewport not {...props} />
}

export default HideAt
