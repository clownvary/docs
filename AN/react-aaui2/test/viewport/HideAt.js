import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import MediaQuery from 'react-responsive'

import HideAt from '../../src/viewport/HideAt'

beforeAll(() => {
  MediaQuery.defaultProps.values = { width: 900 }
})

afterAll(() => {
  MediaQuery.defaultProps.values = {}
})

const setup = (props = {}) => {
  const Component = (
    <HideAt {...props}>
      <p>Hide Me</p>
    </HideAt>
  )
  const wrapper = mount(Component)

  return {
    Component,
    wrapper,
  }
}

it('should render without errors', () => {
  const { wrapper } = setup()

  expect(wrapper).toBeTruthy()
})

it("should render right thing and don't change unexpected", () => {
  const { Component, wrapper } = setup({ smAndAbove: true })
  const tree = renderer.create(Component).toJSON()

  expect(tree).toMatchSnapshot()
  expect(wrapper.find(MediaQuery).props()).toMatchSnapshot()
})
