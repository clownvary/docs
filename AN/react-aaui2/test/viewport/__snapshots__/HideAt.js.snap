// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render right thing and don't change unexpected 1`] = `null`;

exports[`should render right thing and don't change unexpected 2`] = `
Object {
  "children": <p>
    Hide Me
  </p>,
  "query": "not screen and (min-width: 768px)",
  "values": Object {
    "width": 900,
  },
}
`;
